test/Pass/alive0-fail.ll

```diff
@@ -0,0 +1,35 @@
+; REQUIRES: solver
+
+; RUN: %llvm-as -o %t %s
+; RUN: %opt -load %pass -souper -dce %solver -souper-infer-inst -souper-synthesis-comps=and,or,ne,xor,const,add,sub,ashr -S -o - %s | %FileCheck %s
+;XFAIL: *
+
+define i64 @alive0_f0(i64) local_unnamed_addr #0 {
+  %c = icmp slt i64 -1, %0
+  %r = select i1 %c, i64 -8494242232709579858, i64 -3671342357661587618
+  ret i64 %r
+
+  ;CHECK-NOT: icmp slt i64 -1, %0
+  ;CHECK-NOT: select i1 %c, i64 -8494242232709579858, i64 -3671342357661587618
```

Comment:
instead of CHECK-NOT, please check for the expected transformation results (in this one and others)

Reply:
Fixed